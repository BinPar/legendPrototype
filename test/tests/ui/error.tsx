import React from 'react';
import { mount } from 'enzyme';
import Error from '../../../pages/_error';

/**
 * Checks that the error page renders correctly
 */
test('Home Page', () => {
  // We test the res based version of the error page
  const responseError = Error.getInitialProps({ res: { statusCode: 500 } });

  // Renders the page
  const errorPageWithResponse = mount(<Error {...responseError} />);

  // We check that the content of the page is not changed naming the snapshot
  // because we have many models for the error page
  expect(errorPageWithResponse).toMatchSnapshot('errorPageWithResponse');

  // We test the err based version of the error page
  const errError = Error.getInitialProps({ err: { statusCode: 501 } });

  // Renders the page
  const errorPageWithErr = mount(<Error {...errError} />);

  // We check that the content of the page is not changed naming the snapshot
  // because we have many models for the error page
  expect(errorPageWithErr).toMatchSnapshot('errorPageWithErr');

  // We test the err based version of the error page
  const emptyError = Error.getInitialProps({});

  // Renders the page
  const errorPageWithNoInfo = mount(<Error {...emptyError} />);

  // We check that the content of the page is not changed naming the snapshot
  // because we have many models for the error page
  expect(errorPageWithNoInfo).toMatchSnapshot('errorPageWithNoInfo');

  // Renders the page
  const clientSideError = mount(<Error />);

  // We check that the content of the page is not changed naming the snapshot
  // because we have many models for the error page
  expect(clientSideError).toMatchSnapshot('clientSideError');
});
