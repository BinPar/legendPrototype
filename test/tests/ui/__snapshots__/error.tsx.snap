// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page: clientSideError 1`] = `
<Error>
  <Head>
    <Component
      handleStateChange={null}
      inAmpMode={false}
      reduceComponentsToState={[Function]}
    />
  </Head>
  <h1>
    An error occurred on client
  </h1>
</Error>
`;

exports[`Home Page: errorPageWithErr 1`] = `
<Error
  statusCode={501}
>
  <Head>
    <Component
      handleStateChange={null}
      inAmpMode={false}
      reduceComponentsToState={[Function]}
    />
  </Head>
  <h1>
    An error 501 occurred on server
  </h1>
</Error>
`;

exports[`Home Page: errorPageWithNoInfo 1`] = `
<Error
  statusCode={404}
>
  <Head>
    <Component
      handleStateChange={null}
      inAmpMode={false}
      reduceComponentsToState={[Function]}
    />
  </Head>
  <h1>
    An error 404 occurred on server
  </h1>
</Error>
`;

exports[`Home Page: errorPageWithResponse 1`] = `
<Error
  statusCode={500}
>
  <Head>
    <Component
      handleStateChange={null}
      inAmpMode={false}
      reduceComponentsToState={[Function]}
    />
  </Head>
  <h1>
    An error 500 occurred on server
  </h1>
</Error>
`;
